restore_map_to_defaults

dogayalcin opened the pull request:
Add function to replace map with default when player press M
Closes #37

KarinaKuznecova commented on src/main/java/base/navigationservice/KeyboardListener.java:
```diff
@@ -26,6 +26,9 @@ public void keyPressed(KeyEvent event) {
         if (keys[KeyEvent.VK_CONTROL] && keys[KeyEvent.VK_S]) {
             game.handleCTRLandS();
         }
+        if(keys[KeyEvent.VK_M]){ //press M
```
Hi. Thank you for participating in my project :)
I have a few comments. 
1. Let's assign another key, I think H suits here better (as H for Home), since this function will teleport player to Home map. And M will be reserved to show minimap later

KarinaKuznecova commented on src/main/java/base/Game.java:
```diff
@@ -341,4 +341,18 @@ public RenderHandler getRenderer() {
     }
+
+    //replace map to default map
+    public void defualtMap(){       
```
Typo here, should be 'defaultMap'

KarinaKuznecova commented on src/main/java/base/Game.java:
```diff
@@ -341,4 +341,18 @@ public RenderHandler getRenderer() {
+
+       logger.info("Default game map loaded");
+       
+       renderer.adjustCamera(this, player);
```
Would be great to make player appear in the center of map. Now player appears on the same spot as in old map, but if old map was bigger than default - it can cause problems, like player will appear outside of map borders.
I would suggest to create method in Player.java like teleportToCenter(), inside calculate X and Y of the center of current map and place player there. 
If that's too much, let me know, I can create separate issue for that and fix it later

KarinaKuznecova commented on src/main/java/base/navigationservice/KeyboardListener.java:
```diff
@@ -26,6 +26,9 @@ public void keyPressed(KeyEvent event) {
             game.handleCTRLandS();
         }
+        if(keys[KeyEvent.VK_M]){ //press M
+            game.defualtMap();    //replace map with default
```
one more thing: 
comments can be removed from code as long as we have meaningful names. So instead of leaving a comment 'replace map with default' we can name a method 'replaceMapWithDefault'

dogayalcin commented:
I changed the key from M to H,
I removed comments and change the method name (replaceMapWithDefault) 
I added a new method (teleportToCenter()), this method calculate the center of current map and place player there.

KarinaKuznecova commented:
Thanks, looks good now.
